tensorflow/python/ops/array_grad.py

```diff
@@ -115,6 +115,8 @@ def _ExtractInputShapes(inputs):
     if context.executing_eagerly() or isinstance(concat_dim, ops.EagerTensor):
       # Using mod here for convenience since concat_dim is already verified
       # in concat implementation to be within the allowed [-rank, rank) range.
+      if(input_value[0].shape==(1,)):
```

Comment:
Wouldn't this cause problems if it shape was (), not (1,)? Not sure I see what this is fixing; consider adding an explanatory comment.